src/classes/CON_CannotDelete_TEST.cls

```diff
@@ -35,39 +35,1368 @@
 * @description Tests for CON_CannotDelete_TDTM.
 */
 @isTest
-public with sharing class CON_CannotDelete_TEST {
+private with sharing class CON_CannotDelete_TEST {
+    /*********************************************************************************************************
+    * @description Test method to test if Prevent_Contact_Deletion__c is enabled in Hierarchy Settings, and
+    * Contact has an Address record associated to it, that it cannot be deleted.
+    */
+    @isTest
+    private static void conCannotDeleteWithAddress() {
+        UTIL_CustomSettingsFacade.getSettingsForTests(new Hierarchy_Settings__c
+                                                         (Account_Processor__c = UTIL_Describe_API.getHhAccRecTypeID(),
+                                                          Prevent_Contact_Deletion__c = true));
+
+        UTIL_UnitTestData_TEST.ContactsWithAddrs consAccsAddrs =
+            UTIL_UnitTestData_TEST.createTestContactsAddrs(2);
+
+        List<Contact> returnNewContacts = consAccsAddrs.contacts;
+
+
+        List<Contact> queryContacts = [SELECT Id
+                                       FROM Contact
+                                       WHERE Id IN :returnNewContacts];
 
+        Test.startTest();
+        Database.DeleteResult[] results = Database.delete(queryContacts, false);
```

Comment:
this comment goes for all the tests. Do we know if one failure is stopping the whole batch of deletes? If so, probably not a big deal, but if not, we may want to have some tests that test situations where when we delete some contacts, a few don't have any blockers and others do.

Reply:
@lifewithryan  good point. Added a new test that test if there are 2 records, 1 Contact has child related records, and 1 without and if I delete both that the Contact with child related records will not be deleted and the one without child related records will be deleted.